ironfish/src/wallet/wallet.ts

```diff
@@ -680,6 +681,27 @@ export class Accounts {
     try {
       this.assertHasAccount(sender)
 
+      // check if the chain data is fully synced
+      let headHash = this.headHashes.get(sender.id)
+      if (!headHash) {
+        headHash = await sender.getHeadHash()
+      }
+
+      if (!headHash) {
+        throw new ResponseError(`No headhash found for this account.`, ERROR_CODES.ERROR, 400)
+      }
+
+      const header = await this.chain.getHeader(headHash)
+      Assert.isNotNull(header, `Missing block header for hash '${headHash.toString('hex')}'`)
+
+      if (header.timestamp.valueOf() < Date.now() - MAX_SYNCED_AGE_MS) {
```

Comment:
This should not be based on time, but based on if we processed the chain or not in the chain processor.

We should just check if the account head is scanned up to the chain head. It's safe to create transactions when the chain is not synced, but not safe to create them when we haven't processed the chain.

In that case, it can be 1 single line to check the account head sequence vs the chain head sequence, or just that they match if we want to be more strict.

Reply:
How do we know if we haven't processed the chain when they create the transaction? @NullSoldier

Reply:
It's when the account head matches the chain head, which is what isAccountUpToDate does